plugin/src/Internal/Cache.hs

```diff
@@ -28,28 +29,36 @@ import GHC.Types.Name.Cache (NameCache (..), OrigNameCache)
 import GHC.Types.Unique.DFM (plusUDFM)
 import GHC.Types.Unique.FM (UniqFM, minusUFM, nonDetEltsUFM, sizeUFM)
 import GHC.Types.Unique.Supply (initUniqSupply)
-import GHC.Unit.Env (UnitEnv (..))
+import GHC.Unit.Env (HomeUnitEnv (..), HomeUnitGraph, UnitEnv (..), unitEnv_union)
 import GHC.Unit.External (ExternalUnitCache (..), initExternalUnitCache)
-import GHC.Unit.Module.Env (emptyModuleEnv, moduleEnvKeys, plusModuleEnv)
+import GHC.Unit.Finder (InstalledFindResult (..))
+import GHC.Unit.Finder.Types (FinderCache (..))
+import GHC.Unit.Module.Env (InstalledModuleEnv, emptyModuleEnv, moduleEnvKeys, plusModuleEnv)
+import GHC.Unit.Module.Graph (ModuleGraph, unionMG)
 import qualified GHC.Utils.Outputable as Outputable
 import GHC.Utils.Outputable (SDoc, comma, doublePrec, fsep, hang, nest, punctuate, text, vcat, ($$), (<+>))
 import Internal.Log (Log, logd)
 import System.Environment (lookupEnv)
 
 #if __GLASGOW_HASKELL__ >= 911
 
-import Data.IORef (IORef, newIORef, readIORef)
+import Data.IORef (IORef, newIORef)
 import qualified Data.Map.Lazy as LazyMap
 import GHC.Fingerprint (Fingerprint, getFileHash)
 import GHC.IORef (atomicModifyIORef')
 import GHC.Unit (InstalledModule, extendInstalledModuleEnv, lookupInstalledModuleEnv)
 import GHC.Unit.Finder (FinderCache (..), InstalledFindResult (..))
-import GHC.Unit.Module.Env (InstalledModuleEnv, emptyInstalledModuleEnv)
 import GHC.Utils.Panic (panic)
 
 #else
 
-import GHC.Unit.Finder (FinderCache, initFinderCache)
+import GHC.Unit.Finder (initFinderCache)
+
+#endif
+
+#if defined(MWB)
```

Comment:
What is this CPP pragma for? Would you explain this in the code?

Reply:
ah, it might not even be needed anymore, I'll check. But it's intended to adapt to the specific patchset in mwb.